nano/node/node.cpp

```diff
@@ -1089,8 +1089,11 @@ void nano::node::add_initial_peers ()
 
 void nano::node::block_confirm (std::shared_ptr<nano::block> block_a)
 {
-	active.insert (block_a, false);
-	network.broadcast_confirm_req (block_a);
+	auto election = active.insert (block_a, false);
+	if (election.second)
+	{
+		election.first->transition_active ();
```

Comment:
Seems not protected & can use/modify some timepoints in election

Reply:
Pushed some new commits. This is handled by making some timepoints atomic (confirm_req timepoint is protected by active mutex).

Reply:
Re-done with a mutex since atomic timepoints are not a guaranteed support.